### DISCUSSION TITLE
Remote backup with Rclone to S3 bucket

### DISCUSSION THREAD

**Discussion by mpanhans** [nextcloud/all-in-one#1973] [Questions] [answered]
I use AIO's built-in local backup feature, and then periodically I use rclone to copy that local archive to a remote s3 bucket. My question is just basically checking whether the steps I take are fine or are there any improvements. I do this when I know there is no local backup occurring (and no lock file). 

I would prefer not to run rclone as sudo, but because the archives are owned by root user and root group, my user's rclone can’t access these files. So I run sudo rclone and need to specify the rclone config file that the user set up.

Set up: I first configured rclone to connect to a remote Backblaze B2 bucket using steps given by B2 here: https://rclone.org/b2/. Then I use the commands below whenever I want to copy a local backup to the remote storage.

```
# Mount the archives to /tmp/borg.
sudo mkdir -p /tmp/borg && sudo borg mount -o allow_other "/mnt/backup/borg" /tmp/borg

# List all archives
sudo borg list "/mnt/backup/borg"

# Pick most recent one and check its size
sudo du -sh /tmp/borg/yyyymmdd_tttttt-nextcloud-aio

# rclone to b2, using --progress flag for stats and summary at the end
sudo rclone copy /tmp/borg/yyyymmdd_tttttt-nextcloud-aio b2:nextcloud-backup-aio/yyyymmdd_tttttt-nextcloud-aio --progress --config /home/user/.config/rclone/rclone.conf

# And when finished, unmount the backup archive
sudo umount /tmp/borg
```

**Comment by szaimen** [comment, marked as answer]
Hi, why don't you simply sync the complete borg archive to the remote location? Advantage: things are encrypted, you keep all versions, things are smaller because only changes are kept, etc.

**Reply by mpanhans** [reply]
Good suggestion, thanks, that is exactly the kind of feedback I was looking for!
